src/main/java/com/optiva/tools/load/AsyncPublishingConnectionManager.java

```diff
@@ -0,0 +1,78 @@
+package com.optiva.tools.load;
+
+import com.optiva.tools.EventProperties;
+import com.optiva.tools.addevents.NatsConfiguration;
+import com.optiva.tools.addevents.NatsReaderConfiguration;
+import io.nats.client.Connection;
+import io.nats.client.JetStream;
+import io.nats.client.JetStreamOptions;
+import io.nats.client.Message;
+import io.nats.client.Nats;
+import io.nats.client.Options;
+import io.nats.client.PublishOptions;
+import io.nats.client.api.PublishAck;
+import io.nats.client.impl.NatsMessage;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+
+public enum AsyncPublishingConnectionManager {
+    INSTANCE;
+
+    private final Connection natsConnection;
+    private final JetStream jetStream;
+    private final NatsConfiguration natsConfiguration;
+
+    AsyncPublishingConnectionManager()
+    {
+        natsConfiguration = createConfiguration();
+        natsConnection = createConnection();
+        jetStream = createJetstream();
+    }
+
+    public static AsyncPublishingConnectionManager getInstance()
+    {
+        return INSTANCE;
+    }
+
+    public CompletableFuture<PublishAck> publishAysnc(ByteArrayOutputStream event, String subject, String streamName) {
+        Message msg = NatsMessage.builder().subject(subject).data(event.toByteArray()).build();
+        PublishOptions publishOptions = PublishOptions.builder().streamTimeout(Duration.ofSeconds(30)).stream(streamName).build();
+        return jetStream.publishAsync(msg, publishOptions);
```

Comment:
do we need to check the state of the connection to make sure its connected or not in error?
The re-creation of the single instance would beed to be thread safe
Might need to generate the jetstream from any new connection

Reply:
so from JNM, jetstream is meant to be reused, and the connection itself is mean to be threadsafe and handle reconnects automatically, we probably need some solid testing around this, like don't have any explicit reconnect code, but drop pods let and right and make sure things do reconnect.

I went the way of the enum as it is a guaranteed singleton and threadsafe

Reply:
fair enough point. So re-use the jetstream and it should also re-connect